editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -551,14 +551,50 @@ impl MessageHandler<DocumentMessage, (u64, &InputPreprocessorMessageHandler, &Pe
 					.into(),
 				);
 			}
-			NudgeSelectedLayers { delta_x, delta_y } => {
+			NudgeSelectedLayers {
+				delta_x,
+				delta_y,
+				resize,
+				resize_opposite_corner,
+			} => {
 				self.backup(responses);
+				let keyboard = &ipp.keyboard;
+				let is_scale = keyboard.key(resize);
+				let is_bottom_right = !keyboard.key(resize_opposite_corner);
 				for path in self.selected_layers().map(|path| path.to_vec()) {
-					let operation = DocumentOperation::TransformLayerInViewport {
-						path,
-						transform: DAffine2::from_translation((delta_x, delta_y).into()).to_cols_array(),
-					};
-					responses.push_back(operation.into());
+					if is_scale {
+						if let Ok(Some(existing_bounds)) = self.document_legacy.viewport_bounding_box(&path, &render_data) {
+							let [existing_top_left, existing_bottom_right] = [existing_bounds[0], existing_bounds[1]];
```

Comment:
`[existing_bounds[0], existing_bounds[1]]` is redundant, you're converting `existing_bounds` into itself.